servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -520,6 +549,18 @@ private static StreamingHttpConnectionFilterFactory appendConnectionFilter(
     public DefaultSingleAddressHttpClientBuilder<U, R> appendClientFilter(
             final StreamingHttpClientFilterFactory factory) {
         requireNonNull(factory);
+        // TODO 0.42 remove this extra test with class-name
+        if (factory instanceof RetryingHttpRequesterFilter ||
+                factory.getClass().getSimpleName().contains("RetryingHttpRequesterFilter")) {
```

Comment:
Can you clarify why the class-name check is also necessary? If it's for `RetryingHttpRequesterFilter` from utils, consider using the full name or check the instance/class, http-utils are visible in http-netty.